src/lib/djerba/helpers/expression_helper/helper.py

```diff
@@ -0,0 +1,122 @@
+        if wrapper.get_my_string(self.TCGA_CODE_KEY) == 'NULL': # TODO implement is_null()
+            wrapper.set_my_param(self.TCGA_CODE_KEY, project)
+        samples = sample_name_container() # empty container; TODO sample names in INI
+        reader = provenance_reader(fpr_path, project, donor, samples)
+        rsem_genes_results = reader.parse_gep_path()
+        wrapper.set_my_param(self.RSEM_GENES_RESULTS_KEY, rsem_genes_results)
+        return wrapper.get_config()
+
+    def extract(self, config):
+        wrapper = self.get_config_wrapper(config)
+        results = wrapper.get_my_string(self.RSEM_GENES_RESULTS_KEY)
+        gep_reference = wrapper.get_my_string(self.GEP_REFERENCE_KEY)
+        tumour_id = wrapper.get_core_string(self.TUMOUR_ID)
+        gep_abs_path = self.preprocess_gep(results, gep_reference, tumour_id)
+        # Run the R script with output to the workspace
+        cmd = [
+            'Rscript',
+            os.path.join(self.get_module_dir(), 'find_expression.R'),
+            '--enscon ', wrapper.get_my_string(self.ENSCON_KEY),
+            '--genelist', wrapper.get_my_string(self.GENE_LIST_KEY),
+            '--gepfile', gep_abs_path,
+            '--outdir', self.workspace.get_work_dir(),
+            '--tcgadata', wrapper.get_my_string(self.TCGA_DATA_KEY),
+            '--tcgacode', wrapper.get_my_string(self.TCGA_CODE_KEY)
+        ]
+        self.logger.debug("Rscript command: "+" ".join(cmd))
+        #subprocess_runner(self.log_level, self.log_path).run(cmd)
+
+    def preprocess_gep(self, gep_path, gep_reference, tumour_id):
+        """
+        Apply preprocessing to a GEP file; write results to tmp_dir
+        CGI-Tools constructs the GEP file from scratch, but only one column actually varies
+        As a shortcut, we insert the first column into a ready-made file
+        TODO This is a legacy CGI-Tools method, is there a cleaner way to do it?
+        TODO Should GEP_REFERENCE (list of past GEP results) be updated on a regular basis?
+        """
+        # read the gene id and FPKM metric from the GEP file for this report
+        fkpm = {}
+        with open(gep_path) as gep_file:
+            reader = csv.reader(gep_file, delimiter="\t")
+            for row in reader:
+                try:
+                    fkpm[row[self.GENE_ID]] = row[self.FPKM]
+                except IndexError as err:
+                    msg = "Incorrect number of columns in GEP row: '{0}'".format(row)+\
+                          "read from '{0}'".format(gep_path)
+                    self.logger.error(msg)
+                    raise RuntimeError(msg) from err
+        # insert as the second column in the generic GEP file
+        ref_path = gep_reference
+        out_file_name = 'gep.txt'
+        with \
+             gzip.open(ref_path, 'rt', encoding=constants.TEXT_ENCODING) as in_file, \
+             self.workspace.open_file(out_file_name, 'wt') as out_file:
```

Comment:
`open_file` is a method of `workspace` -- it takes a path relative to the workspace directory (which can be only a file name), not a full path